test/unit/test_exports/test_trap_density_xdmf_export.py

```diff
@@ -2,6 +2,7 @@
 from fenics import *
 import sympy as sp
 from pathlib import Path
+import numpy as np
 
 
 def test_trap_density_xdmf_export_intergration_with_simultion(tmpdir):
```

Comment:
This test only tests the feature with a monomaterial so it doesn't prove the new feature works as expected.
I would create a 1D domain with 2 subdomains and put a trap only in one of the domains.